retroshare-gui/src/gui/chat/ChatWidget.cpp

```diff
@@ -393,30 +430,99 @@ void ChatWidget::init(const ChatId &chat_id, const QString &title)
 				if (time(nullptr) <= historyIt->recvTime+2)
 					continue;
 
-				QString name;
-				if (chatId.isLobbyId() || chatId.isDistantChatId())
-				{
-					RsIdentityDetails details;
-					time_t start = time(nullptr);
-					while (!rsIdentity->getIdDetails(RsGxsId(historyIt->peerName), details))
-					{
-						std::this_thread::sleep_for(std::chrono::milliseconds(10));
-						if (time(nullptr)>start+2)
-						{
-							std::cerr << "ChatWidget History haven't found Id Details and have wait 1 sec for it." << std::endl;
-							break;
+				const std::string
```

Comment:
This is bad because std::string will truncate the data to fir first 0. You'd better use this:
std::string str = historyIt->chatPeerId.toStdString();